### PR TITLE
Add tests for checking invalid bib files error messages in parser

### PR DESCRIPTION
**Pull request by Siedlerchr**
Follow up from #10809

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/test/java/org/jabref/logic/importer/ParserResultTest.java [hunk 1]
```diff
@@ -48,4 +54,80 @@ public void doesNotHaveDuplicateWarnings() {
...
+                  journal   = "Journal Name",
+                  year      = "2024",
+                  pages     = "1-10",
+                  publisher = "Publisher Name"
+                }
+                """;
+        Path tempFile = tmpDir.resolve("invalidBibTex.bib");
+        Files.write(tempFile, bibtexEntry.getBytes());
+        ParserResult parserResult = OpenDatabase.loadDatabase(tempFile, mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor());
+        assertEquals("Line 1: Found corrupted citation key  (contains whitespaces).", parserResult.warnings().getFirst());
```
Comment (calixtus) [inline src/test/java/org/jabref/logic/importer/ParserResultTest.java:74]: Two whitespaces before the brackets on purpose?

File: src/test/java/org/jabref/logic/importer/ParserResultTest.java [hunk 2]
```diff
@@ -48,4 +54,80 @@ public void doesNotHaveDuplicateWarnings() {
...
+               year      = "2024",
+               pages     = "1-10",
+               publisher = "Publisher Name"
+             }
+            """;
+
+        Path tempFile = tmpDir.resolve("invalidBibTex.bib");
+        Files.write(tempFile, bibtexEntry.getBytes());
+        ParserResult parserResult = OpenDatabase.loadDatabase(tempFile, mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor());
+        assertEquals("Line 2: Found corrupted citation key .", parserResult.warnings().getFirst());
```
Comment (calixtus) [inline src/test/java/org/jabref/logic/importer/ParserResultTest.java:112]: Whitespace before the dot on purpose?

Comment (Siedlerchr) [reply src/test/java/org/jabref/logic/importer/ParserResultTest.java:112]: This is the %1 parameter for the citation key: 
Line\ %0\:\ Found\ corrupted\ citation\ key\ %1.=Line %0: Found corrupted citation key %1.

File: src/test/java/org/jabref/logic/importer/ParserResultTest.java (hunk 1 above)
Comment (Siedlerchr) [reply src/test/java/org/jabref/logic/importer/ParserResultTest.java:74]: same as aboec

**Pull request merged**